beamer/agent.py

```diff
@@ -79,44 +79,37 @@ def __init__(self, config: Config):
         )
         self._event_processor = EventProcessor(self.context)
 
-        self._contract_monitor_l2a = ContractEventMonitor(
-            "RequestManager",
-            request_manager,
-            l2a_contracts_info["RequestManager"].deployment_block,
-            self._event_processor.add_events,
-            self._event_processor.mark_sync_done,
+        self._contract_monitor_l2a = EventMonitor(
+            web3=w3_l2a,
+            contracts=[request_manager, resolution_registry],
+            deployment_block=min(
+                l2a_contracts_info["RequestManager"].deployment_block,
+                l2a_contracts_info["ResolutionRegistry"].deployment_block,
+            ),
+            on_new_events=self._event_processor.add_events,
+            on_sync_done=self._event_processor.mark_sync_done,
         )
 
-        self._contract_monitor_l2a_registry = ContractEventMonitor(
-            "ResolutionRegistry",
-            resolution_registry,
-            l2a_contracts_info["ResolutionRegistry"].deployment_block,
-            self._event_processor.add_events,
-            self._event_processor.mark_sync_done,
-        )
-
-        self._contract_monitor_l2b = ContractEventMonitor(
-            "FillManager",
-            fill_manager,
-            l2b_contracts_info["FillManager"].deployment_block,
-            self._event_processor.add_events,
-            self._event_processor.mark_sync_done,
+        self._contract_monitor_l2b = EventMonitor(
```

Comment:
Nitpick: consider renaming the instance variable to `_event_monitor_l2b` to be consistent with the new type name.